propagation.go

```diff
@@ -169,6 +164,107 @@ func (p *textMapPropagator) Extract(abstractCarrier interface{}) (SpanContext, e
 	return ctx, nil
 }
 
+type zipkinB3Propagator struct {
+	*tracer
+	*coder
+}
+
+func (p *zipkinB3Propagator) Inject(
+	sc jaeger.SpanContext,
+	abstractCarrier interface{},
+) error {
+	textMapWriter, ok := abstractCarrier.(opentracing.TextMapWriter)
+	if !ok {
+		return opentracing.ErrInvalidCarrier
+	}
+
+	textMapWriter.Set("x-b3-traceid", p.Encode(strconv.FormatUint(sc.TraceID(), 16)))
+	if sc.ParentID() != 0 {
+		textMapWriter.Set("x-b3-parentspanid", p.Encode(strconv.FormatUint(sc.ParentID(), 16)))
+	}
+	textMapWriter.Set("x-b3-spanid", p.Encode(strconv.FormatUint(sc.SpanID(), 16)))
+	if sc.IsSampled() {
+		textMapWriter.Set("x-b3-sampled", p.Encode("1"))
+	} else {
+		textMapWriter.Set("x-b3-sampled", p.Encode("0"))
+	}
+	return nil
+}
+
+func (p *zipkinB3Propagator) Extract(abstractCarrier interface{}) (SpanContext, error) {
+	textMapReader, ok := abstractCarrier.(opentracing.TextMapReader)
+	if !ok {
+		return emptyContext, opentracing.ErrInvalidCarrier
+	}
+	var traceID uint64
+	var spanID uint64
+	var parentID uint64
+	sampled := false
+	err := textMapReader.ForeachKey(func(rawKey, value string) error {
+		key := strings.ToLower(rawKey) // TODO not necessary for plain TextMap
+		var err error
+		if key == "x-b3-traceid" {
+			traceID, err = strconv.ParseUint(p.Decode(value), 16, 64)
+		} else if key == "x-b3-parentspanid" {
+			parentID, err = strconv.ParseUint(p.Decode(value), 16, 64)
+		} else if key == "x-b3-spanid" {
+			spanID, err = strconv.ParseUint(p.Decode(value), 16, 64)
+		} else if key == "x-b3-sampled" && value == "1" {
+			sampled = true
+		}
+		return err
+	})
+
+	if err != nil {
+		p.tracer.metrics.DecodingErrors.Inc(1)
+		return emptyContext, err
+	}
+	if traceID == 0 {
+		return emptyContext, opentracing.ErrSpanContextNotFound
+	}
+	return NewSpanContext(traceID, spanID, parentID, sampled, nil), nil
+}
+
+func newJaegerHTTPHeaderPropagator(tracer *tracer) *jaegerPropagator {
+	return &jaegerPropagator{
+		coder:  httpHeaderCoder,
+		tracer: tracer,
+	}
+}
+
+func newZipkinB3HTTPHeaderPropagator(tracer *tracer) *zipkinB3Propagator {
+	return &zipkinB3Propagator{
+		coder:  httpHeaderCoder,
+		tracer: tracer,
+	}
+}
+
+func newJaegerTextMapPropagator(tracer *tracer) *jaegerPropagator {
```

Comment:
similar to package name, we are in jaeger namespace already, and don't need `Jaeger` prefix for the classes